src/script/calling/CallMessageBuilder.js

```diff
@@ -19,107 +19,102 @@
-  };
+/**
+ * Create additional payload.
+ *
+ * @param {string} conversationId - ID of conversation
+ * @param {string} selfUserId - ID of self user
+ * @param {string} [remoteUserId] - Optional ID of remote user
+ * @param {string} [remoteClientId] - Optional ID of remote client
+ * @returns {{conversationId: string, remoteClientId: string, remoteUserId: *, time: string, userId: string}} Additional payload
+ */
+const _createPayload = (conversationId, selfUserId, remoteUserId, remoteClientId) => {
+  return {conversationId, remoteClientId, remoteUserId, time: new Date().toISOString(), userId: selfUserId};
+};
 
-  /**
-   * Create properties payload for call events.
-   *
-   * @param {Object} selfState - Current self state
-   * @param {Object} additionalPayload - Optional additional payload to be added
-   * @param {boolean} [videoStateOverwrite] - Forces the videosend property to be this value instead of the one in the selfState
-   * @returns {Object} call message props object
-   */
-  const _createPropSync = (selfState, additionalPayload, videoStateOverwrite) => {
-    const payload = {};
-    const {audioSend: audioState, videoSend: videoState, screenSend: screenState} = selfState;
-    const videoSend = _.isBoolean(videoStateOverwrite) ? videoStateOverwrite : videoState();
-
-    payload.properties = {
-      audiosend: `${audioState()}`,
-      screensend: `${screenState()}`,
-      videosend: `${videoSend}`,
-    };
-
-    return additionalPayload ? Object.assign(payload, additionalPayload) : payload;
+/**
+ * Create properties payload for call events.
+ *
+ * @param {Object} selfState - Current self state
+ * @param {Object} additionalPayload - Optional additional payload to be added
+ * @param {boolean} [videoStateOverwrite] - Forces the videosend property to be this value instead of the one in the selfState
+ * @returns {Object} call message props object
+ */
+const _createPropSync = (selfState, additionalPayload, videoStateOverwrite) => {
+  const payload = {};
+  const {audioSend: audioState, videoSend: videoState, screenSend: screenState} = selfState;
+  const videoSend = _.isBoolean(videoStateOverwrite) ? videoStateOverwrite : videoState();
+
+  payload.properties = {
+    audiosend: `${audioState()}`,
+    screensend: `${screenState()}`,
+    videosend: `${videoSend}`,
   };
 
-  return {
-    buildCancel: _buildCancel,
-    buildGroupCheck: _buildGroupCheck,
-    buildGroupLeave: _buildGroupLeave,
-    buildGroupSetup: _buildGroupSetup,
-    buildGroupStart: _buildGroupStart,
-    buildHangup: _buildHangup,
-    buildPropSync: _buildPropSync,
-    buildReject: _buildReject,
-    buildSetup: _buildSetup,
-    buildUpdate: _buildUpdate,
-    createPayload: _createPayload,
-    createPropSync: _createPropSync,
-  };
-})();
+  return additionalPayload ? Object.assign(payload, additionalPayload) : payload;
+};
+
+export const CallMessageBuilder = {
```

Comment:
Why not also refactor this by removing the unnecessary renaming?

Reply:
Fair point :+1: